provider/postgis/postgis.go

```diff
@@ -199,14 +221,49 @@ func NewTileProvider(config dict.Dicter) (provider.Tiler, error) {
 		return nil, err
 	}
 
+	// check if provider layer are automatically generated
+	var autoTables []string
+	if auto {
+		// get auto config settings
+		// *NOTE: I'm guessing this should be a map, not a map slice (we only want 1 auto config per provider) but couldn't get it to work
+		autoConfig, err := config.MapSlice(ConfigKeyAutoConfig)
+
+		if err != nil {
+			return nil, err
+		}
+		// generate layers from information_schema of provider
+		autoTables, err = genLayers(p.pool)
+		for i := 0; i < len(autoTables); i++ {
+
+			// this is very hacky. I was having problems creating an empty dict.Dicter
+			defaultLayer, err := config.Map("")
+			if err != nil {
+				log.Println(err)
+			}
+			if len(autoConfig) > 0 {
+				// use auto config details to fill in layer info
+				defaultLayer = autoConfig[0]
+			}
+			// add auto config layer
```

Comment:
Ideally, we could add the layer name to this dict.Dicter element, but I wasn't sure how to do that once the mapSlice has been created. This affects line 261 below